### DISCUSSION TITLE
How can I use the Android 12 Splash Screen API?

### DISCUSSION THREAD

**Discussion by AE1NS** [ionic-team/capacitor#5816] [Q&A]
I am just migrating to capacitor 4 and followed the [migration guide](https://capacitorjs.com/docs/updating/4-0#optional-use-the-new-android-12-splash-screen-api) to use the new Splash Screen API. I would be happy to see any more detailed documentation, how to use Android splash screens with Capacitor in the future, because I just get a white screen after the migration, before the app content shows up.

When I have a look to the capacitors splash screen plugin, the methods are marked with "Doesn't work on launch when using the Android 12 API". So am I right, that switching to the Splash Screen API, deactivates the plugin functionality and just uses the native splash screen, before the app content shows up? If yes, how can I decide when to hide the splash or shouldnt this used in the future (i.e. fastest way to display the app content and show custom loaders or something like that)? Does this mean, that the splash screen plugin is deprecated for Android 31+?

I just did some steps to migrate, like:

build.gradle:
```
implementation "androidx.core:core-splashscreen:$coreSplashScreenVersion"
```

variables.gradle:
```
coreSplashScreenVersion = '1.0.0'
```

v31\styles.xml
```
...
<style name="AppTheme.NoActionBarLaunch" parent="Theme.SplashScreen">
    <item name="android:windowSplashScreenBackground">@color/launchBackground</item>
    <item name="android:windowSplashScreenIconBackgroundColor">@color/launchBackground</item>
    <item name="android:windowSplashScreenAnimatedIcon">@drawable/my_splash</item>
    <item name="android:windowSplashScreenAnimationDuration">3000</item>
    <item name="postSplashScreenTheme">@style/AppTheme.NoActionBar</item>
</style>
...
```
(I downloaded the [example starter kit](https://developer.android.com/guide/topics/ui/splash-screen#splash-screen-resources) and saved the animated example xml as 'my_splash.xml' to the res/drawable folder.)

AndroidManifest:
```
<activity
    android:theme="@style/AppTheme.NoActionBarLaunch"
...
```

MainActivity.java:
```
...
import android.os.Bundle;
import com.getcapacitor.BridgeActivity;
import androidx.core.splashscreen.SplashScreen;

public class MainActivity extends BridgeActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        SplashScreen.installSplashScreen(this);
        super.onCreate(savedInstanceState);
    }
}
```

But I just get a white screen (@color/launchBackground) before the app content shows up. Maybe someone can assist me what I am doing/understanding wrong here :/

Thanks in advance.

**Comment by jcesarmobile** [comment]
The white screen is a known issue of the Android 12 Splash Screen API when the app is launched from Android Studio, it affects Android 31+ devices.
If you close the app and open it from the app icon (as the users will do) then it shows correctly.
It's documented on the plugin
> NOTE: Some issues may exist on SDK >= 31 when using the new API for the splash screen, which pertain to the splash screen only showing when launched from the launcher icon. [Google Issue Tracker](https://issuetracker.google.com/issues/205021357)

You don't have to do anything to use the Splash Screen API, other than having @capacitor/splash-screen plugin installed and having the parent of `AppTheme.NoActionBarLaunch` theme changed to `Theme.SplashScreen`. Splash Screen plugin calls `installSplashScreen`. If you have `launchShowDuration` set to an integer and `launchAutoHide` set to `true` the splash will dismiss when the configured amount of time finish. If you have `launchAutoHide` set to `false` the splash will remain until you call `SplashScreen.hide();`.

The Android 12 Splash Screen API is also used in earlier versions because of the `androidx.core:core-splashscreen` compat library, as long as you have the parent changed to `Theme.SplashScreen`, if you don't change it it will continue using the old splash screen in older devices. But on Android 12+ it will use the Android 12 Splash Screen API despite the theme you use as it's baked into the OS.

**Reply by AE1NS** [reply]
Thank you for your reply.

I already read about that issue, that I have to start the app manually after installing it. Unfortunately my splash logo is still not displayed then, only the set background color. (Samsung Galaxy S20+ with Microsoft Launcher)

I always used cordova-res for my drawable files and after the Cap 4 upgrade I got that issue, without changing anything else (well, at least I hope so ...)

Good to know, that I can still use SplashScreen.hide() feature, makes sense.

Didnt know that the Splash Screen plugin calls installSplashScreen. Thanks for that hint.

Does nobody other has the issue with the missing splash after upgrading? Do you have any further hint?

**Reply by AE1NS** [reply]
Just found out that it seems to be related to third party launchers as Nova Launcher or Microsoft Launcher and looks like the same issue as starting it from Android Studio. I tested it on a Google Pixel 4 with default launcher and its working fine. After changing my S20 to the default OneUI Launcher, its also working.

https://issuetracker.google.com/issues/207386164
